### PR TITLE
Add prettier support

### PR DESCRIPTION
**Pull request by MohammadxAli**
I wanted to make some small adjustments and bug fixes but realized the moment I try to save the file, my editor's prettier tries to auto-format the file which results in a lot of unrelated line changes.

This PR tries to add prettier support to the project so that everyone can follow the same code convention as well as making it easier for outside developers to make improvements to the project. 

The 5f18aef9d439e1a81a3921a15ce53fb326b727e6 commit is a result of running `npm run format` script which was created on 40c6924eba1e60a1b1eb07673159b44aadf69716, and that's why the line changes are too much.

Let me know if you want me to change anything (: hopefully, I can make my other contributions after this gets merged.

### CODE REVIEW

**Comment by mhemrg** [comment]
Hi Mohammad Ali,

Thanks for the contribution.

PR (#71) has just landed on the master branch. You need to rebase your changes or just reformat the master and make your PR again.

**Comment by MohammadxAli** [comment]
Hey @mhemrg!

Ready for another review.

**Comment by mhemrg** [comment]
Thanks, @MohammadxAli.

Is it possible for you to add `husky` and `pretty-quick` to this setup?

And can you please just PR your configuration instead of the formatted version of all these files? It's not possible for us to review all these file changes right now. Then I can just run the `format` command myself and push it directly to the master.

**Comment by MohammadxAli** [comment]
> And can you please just PR your configuration instead of the formatted version of all these files? 

Yeah, definitely that makes much more sense. I was thinking the same.

> Is it possible for you to add husky and pretty-quick to this setup?

Not sure about the `pretty-quick`, but a `husky` hook should be straightforward to add, will let you know once it's done.

**Comment by MohammadxAli** [comment]
@mhemrg Followed by [prettier docs](https://prettier.io/docs/en/precommit.html#option-2-pretty-quickhttpsgithubcomazzpretty-quick), and it's working as expected. 

Gave it a try before making my own commits and it did work. Let me know if it's working on your side.

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  editorconfig: true,
```
Comment (mhemrg) [inline .prettierrc.js:2]: Prettier always reads .editorconfig if it exists. There is no option. 
```suggestion

```

**Comment by mhemrg** [comment]
Thanks again, Mohammad Ali.

**Pull request merged**